core/proto/keytransparency_v1_proto/keytransparency_v1_admin_proto.proto

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/google/keytransparency/core/proto/keytransparency_v1_proto";
+
+package keytransparency.v1.proto;
+
+import "trillian.proto";
+import "crypto/keyspb/keyspb.proto";
+import "proto/keytransparency_v1_proto/keytransparency_v1_proto.proto";
+
+// SetEntriesRequest will update the accounts of the given user_ids to be set to
+// user data if those accounts have key_id in their set of authorized_keys.
+message BatchUpdateEntriesRequest {
+  // users is a map from user_ids to user data.
+  map<string,UserProfile> users = 1;
+
+  // app_id is the application to make this change for.
+  string app_id = 2;
+
+  // key_id is the id of the authorized_public key to use when updating accounts.
```

Comment:
maybe this could be reflected in the field name.  auth_pub_key_id ?

Reply:
Good idea. Done.